src/pages/signup.tsx

```diff
@@ -1,37 +1,53 @@
-import { FC, useEffect } from 'react';
+import { FC, useEffect, useState, useCallback } from 'react';
 import { Auth } from '@/types/enum';
-import { useAppDispatch } from '@/store/store';
-import { setUser } from '@/store/features/userSlice';
 import { getAuth, createUserWithEmailAndPassword } from 'firebase/auth';
 import Form from '@/components/Form';
 import { IFormData } from '@/types/interface';
 import { useRouter } from 'next/router';
 import { useAuth } from '@/hooks/useAuth';
+import { notification } from 'antd';
+import { useTranslation } from 'react-i18next';
 
 const SignUp: FC = () => {
-  const dispatch = useAppDispatch();
+  const [toast, setToast] = useState('');
   const router = useRouter();
   const { isAuth } = useAuth();
+  const [api, contextHolder] = notification.useNotification();
+  const { t } = useTranslation();
+
+  const openErrorNotification = useCallback(() => {
+    api['error']({
+      message: t('errorNotification.messageUp'),
+      description: t('errorNotification.descriptionUp'),
+    });
+  }, [api, t]);
 
   useEffect(() => {
     if (isAuth) {
       router.push('/');
     }
   }, [isAuth, router]);
 
+  useEffect(() => {
+    if (toast === 'error') {
+      openErrorNotification();
+      setToast('');
+    }
+  }, [router, toast, openErrorNotification]);
+
   const handleRegister = (data: IFormData) => {
```

Comment:
```suggestion
  const handleRegister = (data: IFormData) => {
```
This logic can be transferred directly to the form.